scene/gui/code_edit.cpp

```diff
@@ -742,6 +742,48 @@ void CodeEdit::_backspace_internal(int p_caret) {
 	end_complex_operation();
 }
 
+/* Folding Block*/
+bool CodeEdit::_is_fold_blocks_start(int p_line) const {
+	ERR_FAIL_INDEX_V(p_line, get_line_count(), false);
+	for (int i = 0; i < fold_blocks_prefixes.size(); i++) {
+		// Empty string prefix will always return true when begins_with is checked
+		if (fold_blocks_prefixes[i] == "") {
+			continue;
+		}
+		// this line and next  but not previous begins with fold block prefix
+		if (get_line(p_line).begins_with(fold_blocks_prefixes[i]) && get_line(p_line + 1).begins_with(fold_blocks_prefixes[i])) {
+			if (get_line(p_line - 1).begins_with(fold_blocks_prefixes[i])) {
+				return false;
+			}
+			return true;
+		}
+	}
+	return false;
+}
+
+void CodeEdit::set_fold_blocks_enabled(bool p_enabled) {
+	enable_fold_blocks = p_enabled;
+}
+
+bool CodeEdit::is_fold_blocks_enabled() const {
+	return enable_fold_blocks;
+}
+
+void CodeEdit::set_fold_blocks_prefixes(const PackedStringArray &p_prefixes) {
+	fold_blocks_prefixes.clear();
+	for (int i = 0; i < p_prefixes.size(); i++) {
+		const String prefix = p_prefixes[i];
+		if (fold_blocks_prefixes.has(prefix) || prefix == "") {
+			continue;
+		}
+		fold_blocks_prefixes.append(prefix);
+	}
```

Comment:
```suggestion
	for (const String &prefix : p_prefixes) {
		if (prefix.is_empty() || fold_blocks_prefixes.has(prefix)) {
			continue;
		}
		fold_blocks_prefixes.append(prefix);
	}
```
Use a foreach loop, and put the simpler check first